core/domain/platform_feature_services.py

```diff
@@ -149,21 +150,20 @@ def update_feature_flag_rules(
         feature_name, committer_id, commit_message, new_rules)
 
 
-# TODO(#10211): Currently Oppia runs in either of the two modes:
-# dev or prod. There should be another mode 'test' added for QA testing,
-# once it is added, this function needs to be updated to take that into
-# consideration.
 def _get_server_mode() -> platform_parameter_domain.ServerMode:
     """Returns the running mode of Oppia.
 
     Returns:
-        Enum(SERVER_MODES). The server mode of Oppia, dev if Oppia is running
-        in development mode, prod if in production mode.
+        Enum(SERVER_MODES). The server mode of Oppia. This is "dev" if Oppia is
+        running in development mode, "test" if Oppia is running on a test
+        server, and "prod" if Oppia is running in full production mode.
```

Comment:
What about backup server?

Reply:
I'd imagine that'd be 'test' but it's a good question. How do we see feature staging working long-term in the release pipeline when considering local development, test server, backup server, and production?